sql/core/src/test/scala/org/apache/spark/sql/DataFrameFunctionsSuite.scala

```diff
@@ -5179,7 +5179,7 @@ class DataFrameFunctionsSuite extends QueryTest with SharedSparkSession {
       exception = intercept[AnalysisException] {
         df.selectExpr("zip_with(a1, a2, x -> x)")
       },
-      errorClass = "_LEGACY_ERROR_TEMP_2300",
+      errorClass = "INVALID_LAMBDA_FUNCTION_CALL.ARGUMENTS_NUMBER_MISMATCH",
```

Comment:
Let's make it consistent to already existing error classes:
- NUM_COLUMNS_MISMATCH
- WRONG_NUM_ARGS

and make it shorter slightly: `NUM_ARGS_MISMATCH`